python/ql/test/experimental/query-tests/Security/CWE-943/flask_mongoengine_get_db_bad.py

```diff
@@ -0,0 +1,31 @@
+from flask import Flask, request
+from flask_mongoengine import MongoEngine
+import json
+
+app = Flask(__name__)
+db = MongoEngine(app)
+db.init_app(app)
+
+
+class Movie(db.Document):
+    title = db.StringField(required=True)
+    year = db.IntField()
+    rated = db.StringField()
+    director = db.StringField()
+    actors = db.ListField()
+
+
+Movie(title='aa').save()
+Movie(title='bb').save()
+
+
+@app.route("/")
+def home_page():
+    unsanitized_search = request.args['search']
+    json_search = json.loads(unsanitized_search)
+
+    retrieved_db = db.get_db()
+    result = retrieved_db["Movie"].find_one_or_404({'name': json_search})
```

Comment:
This code needs to be fixed. `find_one_or_404` is available in flask_pymongo and not flask_mongoengine.

Reply:
done